sklearn/cross_decomposition/pls_.py

```diff
@@ -38,7 +39,9 @@ def _nipals_twoblocks_inner_loop(X, Y, mode="A", max_iter=500, tol=1e-06,
         # 1.1 Update u: the X weights
         if mode == "B":
             if X_pinv is None:
-                X_pinv = linalg.pinv(X)   # compute once pinv(X)
+                # We use slower pinv2 (same as np.linalg.pinv) for stability
+                # reason
```

Comment:
Only [scipy > 0.12](https://github.com/scipy/scipy/blob/maintenance/0.11.x/scipy/linalg/basic.py#L546) (check the release in this link). Something similar was handled in [omp](https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/linear_model/omp.py#L23). We may do the same.